contracts/UniswapV3Staker.sol

```diff
@@ -156,7 +150,15 @@ contract UniswapV3Staker is IUniswapV3Staker, IERC721Receiver, Multicall {
         emit TokenDeposited(tokenId, from);
 
         if (data.length > 0) {
-            _stakeToken(abi.decode(data, (UpdateStakeParams)));
+            if (data.length == 192) {
```

Comment:
perfect i was going to suggest this. a little fragile until we deploy but probably worth it